src/algebra/group/pi.lean

```diff
@@ -257,3 +257,16 @@ lemma pi.piecewise_div [Π i, has_div (f i)] (s : set I) [Π i, decidable (i ∈
 s.piecewise_op₂ _ _ _ _ (λ _, (/))
 
 end piecewise
+
+section extend
+
+variables {ι : Type u} {η : Type v} (R : Type w) (s : ι → η)
+
+/-- `function.extend s f 1` as a bundled hom. -/
+@[to_additive function.extend_by_zero.add_hom "`function.extend s f 0` as a bundled hom."]
+noncomputable def function.extend_by_one.mul_hom [mul_one_class R] : (ι → R) →* (η → R) :=
```

Comment:
Note that strictly this is a `monoid_hom` not a `mul_hom`, but I'll let you decide whether you care. Frankly even just `extend_by_one_hom` is probably enough, since I don't think there are any other interesting variants.

Reply:
I agree `hom` here is enough.